src/lib.rs

```diff
@@ -244,7 +244,7 @@ impl Debug for SyncCallback {
 }
 
 impl PartialEq for SyncCallback {
-    #[allow(clippy::vtable_address_comparisons)]
+    #[allow(ambiguous_wide_pointer_comparisons)]
```

Comment:
This allow lint can be removed as https://github.com/rust-lang/rust/pull/80505 is merged.